docs/source/best_practices.rst

```diff
@@ -0,0 +1,222 @@
+
+would automatically perform the same sorting and read filtering logic as the previous section.
+TODO: which cases this does and doesn't work
+
+Use broadcast joins when possible
+---------------------------------
+
+Joins and grouped aggregations typically require communication between workers, which can be expensive.
+Broadcast joins can help reduce this communication in the case of joining a small table to a large table by just sending the small table to each partition of the large table.
+However, in Dask-SQL this only works when the small table is a sinlge partition.
+
+For example, if you read in some tables and concatenate them with a ``UNION ALL`` operation
+
+.. code-block:: sql
+    CREATE OR REPLACE TABLE precip AS
+    SELECT
+        station_id,
+        substring("date", 0, 4) as yr,
+        substring("date", 5, 2) as mth,
+        substring("date", 7, 2) as dy,
+        val*1/10*0.0393701 as inches
+    FROM weather_nosplit
+    WHERE type='PRCP'
+
+.. code-block:: sql
+    CREATE OR REPLACE TABLE atlanta_stations WITH (
+        location = '/data/atlanta_stations/*.parquet',
+        gpu=True
+    )
+
+.. code-block:: sql
+    CREATE OR REPLACE TABLE seattle_stations WITH (
+        location = '/data/seattle_stations/*.parquet',
+        gpu=True
+    )
+
+
+.. code-block:: sql
+    CREATE OR REPLACE TABLE city_stations AS
+    SELECT * FROM atlanta_stations
+    UNION ALL
+    SELECT * FROM seattle_stations
+
+you get a new table that has two partitions. Then if you use it in a join
+
+.. code-block:: sql
+    SELECT
+        yr,
+        city,
+        CASE WHEN city='Atlanta' THEN
+            sum(inches)/{atl_stations}
+        ELSE
+            sum(inches)/{seat_stations}
+        END AS inches
+    FROM precip
+    JOIN city_stations
+    ON precip.station_id = city_stations.station_id
+    GROUP BY yr, city
+    ORDER BY yr ASC
+
+Dask-SQL won't perform a broadcast join and will instead perform a traditional join with a corresponding slow computate time.
+However, if you were to repartition the smaller table to a single partition and rerun the operation
+
+.. code-block:: python
+    c.create_table("city_stations", c.sql("select * from city_stations").repartition(npartitions=1))
+
+.. code-block:: sql
+    SELECT
+        yr,
+        city,
+        CASE WHEN city='Atlanta' THEN
+            sum(inches)/{atl_stations}
+        ELSE
+            sum(inches)/{seat_stations}
+        END AS inches
+    FROM precip
+    JOIN city_stations
+    ON precip.station_id = city_stations.station_id
+    GROUP BY yr, city
+    ORDER BY yr ASC
+
+Dask-SQL is able to recognize this as a broadcast join and the result is a significantly faster compute time.
+
+Optimize Partition Sizes for GPUs
+---------------------------------
+File formats like [Apache ORC](https://orc.apache.org/) and [Apache Parquet](https://parquet.apache.org/) are designed so that they can be pulled from disk and be deserialized by CPUs quickly.
```

Comment:
```suggestion
File formats like `Apache ORC <https://orc.apache.org/>`_ and `Apache Parquet <https://parquet.apache.org/>`_ are designed so that they can be pulled from disk and deserialized by CPUs quickly.
```